### ISSUE TITLE
Editor text display is abnormal - character spacing problem

### ISSUE THREAD

**Issue by caijingwen-vincent** [WinMerge/winmerge#468]
Hi folks, there seems to be something wrong with the software displayed on my computer ; Editor text display is abnormal - character spacing problem 。  Here's what it looks like by default

* RenderingMode : GDI
![image](https://user-images.githubusercontent.com/71600126/97692240-7c90f700-1ada-11eb-8eb1-0daeac1c54b6.png)

* DirectWrite Default
![image](https://user-images.githubusercontent.com/71600126/97692283-89154f80-1ada-11eb-933d-fd8f498fbce7.png)

Below is a screenshot of the relevant resolution on my computer
The resolution is 2560*1440 and the font size is 100
![image](https://user-images.githubusercontent.com/71600126/97692683-08a31e80-1adb-11eb-9f5d-b5853f464826.png)

**Comment by sdottaka** [comment]
Could you show us the Font/* part of the settings displayed by selecting [Help] → [Configuration] menu item?

![image](https://user-images.githubusercontent.com/98126/97703025-41e78880-1af3-11eb-801e-a88fd768559d.png)

**Comment by caijingwen-vincent** [comment]
Hi sdottaka,
      Here are the configurations that I found beginning with “Font“ in the file you mentioned 。 


```properties
  Font/CharSet=134
  Font/ClipPrecision=2
  Font/Escapement=0
  Font/FaceName=Microsoft YaHei UI
  Font/Height=-16
  Font/Italic=0
  Font/Orientation=0
  Font/OutPrecision=1
  Font/PitchAndFamily=49
  Font/PointSize=12
  Font/Quality=1
  Font/Specified=0
  Font/StrikeOut=0
  Font/Underline=0
  Font/Weight=400
  FontDirCompare/CharSet=1
  FontDirCompare/ClipPrecision=0
  FontDirCompare/Escapement=0
  FontDirCompare/FaceName=Microsoft YaHei UI
  FontDirCompare/Height=-12
  FontDirCompare/Italic=0
  FontDirCompare/Orientation=0
  FontDirCompare/OutPrecision=0
  FontDirCompare/PitchAndFamily=0
  FontDirCompare/PointSize=9
  FontDirCompare/Quality=0
  FontDirCompare/Specified=0
  FontDirCompare/StrikeOut=0
  FontDirCompare/Underline=0
  FontDirCompare/Weight=400

**Comment by caijingwen-vincent** [comment]
In addition, the software version is the latest 2.16.8; I tried to update to the last version, but there was also this problem

![image](https://user-images.githubusercontent.com/71600126/97719865-89c1dc00-1b02-11eb-8b3b-8e782ac78626.png)

**Comment by sdottaka** [comment]
The Microsoft YaHei UI font seems to be a proportional font.
However, WinMerge only supports fixed-width fonts. Also, since the Microsoft Yahei UI font is a proportional font, it doesn't appear in the WinMerge font selection dialog, so I don't think you can select that font.

Therefore, if you want to use Chinese fonts, choose fixed-width fonts such as NSimSun or SimSum.

![image](https://user-images.githubusercontent.com/98126/97725712-d660e380-1b11-11eb-9438-53d5dccb19f4.png)

**Comment by caijingwen-vincent** [comment]
Hi sdottaka,
I converted the font to Consolas， Now everything is displayed normal . Thank you for your timely support👍 

I tried to switch fonts in Options, but I couldn't find them 。 As you said, it's under the View menu

In addition, I have a suggestion. If the font of the system is not under ``WinMerge font selection dialog (View -> Select Font menu item)`` when the user first use WinMerge, can WinMerge switch the font at WinMerge to the default value like Consolas（Everyone loves it 😆 ） to avoid the problems I met before.  For example, in China, the default font seems to be Microsoft Yahei, and people in other areas may have the same problem

**Comment by sdottaka** [comment]
WinMerge uses the fixed width font name (MIMECPINFO::wszFixedWidthFont) obtained using the IMultiLanguage2::GetCodePageInfo() function as the default font.
For this reason, I thought the above function could not get the name of the proportional font.
However, I was able to reproduce this problem by setting the System locale to "Chinese (Simplified, China)".

In the System locale above, the IMultiLanguage2::GetCodePageInfo() function set the font with the same name as the proportional font (MIMECPINFO::wszProportionalFont) to the fixed-width font name (MIMECPINFO::wszFixedWidthFont). I suspect this is a Windows bug.

For this reason, I have fixed this issue by renaming the default font to "Courier New" if the proportional font name and the fixed-width font name are the same.

Thank you for the report.

![image](https://user-images.githubusercontent.com/98126/97878278-c63f4300-1d61-11eb-8372-1f1ff4566403.png)
